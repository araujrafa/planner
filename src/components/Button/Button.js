import React from 'react';

export default ({text, onClickButton}) => (
  <button onClick={onClickButton} className="btn btn-success">
    {text}
  </button>
)