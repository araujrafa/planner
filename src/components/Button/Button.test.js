import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button component', () => {
  test('Button changes the class when hovered', () => {
    const onClick = jest.fn();
  
    const component = renderer.create(
      <Button text={'Buscar'} onClickButton={onClick}/>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('Button call funtion after click', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Button text={'Buscar'} onClickButton={onClick} />);
  
    wrapper.simulate('click');
  
    expect(onClick).toHaveBeenCalledTimes(1);
  });
})