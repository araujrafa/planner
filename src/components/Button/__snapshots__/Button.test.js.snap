// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button changes the class when hovered 1`] = `
<button
  className="btn btn-success"
  onClick={[Function]}
>
  Buscar
</button>
`;

exports[`Button component Button changes the class when hovered 1`] = `
<button
  className="btn btn-success"
  onClick={[Function]}
>
  Buscar
</button>
`;
